assets/quick-add-bulk.js

```diff
@@ -0,0 +1,159 @@
+      .then((response) => response.text())
+      .then((responseText) => {
+        const html = new DOMParser().parseFromString(responseText, 'text/html');
+        const sourceQty = html.querySelector(`#quick-add-bulk-${this.dataset.id}`);
+        this.innerHTML = sourceQty.innerHTML;
+      })
+      .catch(e => {
+        console.error(e);
+      });
+  }
+
+  updateCart(event) {
+    this.quantity = this.querySelector('quantity-input')
+    // this.quantity.classList.add('loading');
+    this.querySelector('.progress-bar-container').classList.remove('hidden');
+    const body = JSON.stringify({
+      quantity: event.target.value,
+      id: event.target.getAttribute('data-index'),
+      sections: this.getSectionsToRender().map((section) => section.section)
+    });
+
+    fetch(`${routes.cart_change_url}`, { ...fetchConfig('javascript'), ...{ body } })
+      .then((response) => {
+        return response.text();
+      })
+      .then((state) => {
+        const parsedState = JSON.parse(state);
+
+        if (parsedState.description || parsedState.errors) {
+          // const errorElement = document.querySelector(`#Quick-add-bulk-item-error-desktop-${event.target.getAttribute('data-index')}`)
+          // errorElement.classList.remove('hidden')
+          // errorElement.querySelector('.variant-bulk__error-text').innerHTML = parsedState.description
+         // Update errors 
+          return;
+        }
+
+
+        this.renderSections(parsedState);
+
+        publish(PUB_SUB_EVENTS.cartUpdate, { source: "quick-add", cartData: parsedState });
+
+      }).catch((error) => {
+        console.log(error, 'error')
+      })
+  }
+
+  addToCart(event) {
+    this.querySelector('.progress-bar-container').classList.remove('hidden');
+    const body = JSON.stringify({
+      items: [
+        {
+          quantity: parseInt(event.target.value),
+          id: parseInt(this.dataset.id)
+        }
+      ],
+      sections: this.getSectionsToRender().map((section) => section.section)
+    });
+
+    fetch(`${routes.cart_add_url}`, { ...fetchConfig('javascript'), ...{ body } })
+      .then((response) => {
+        return response.text();
+      })
+      .then((state) => {
+        const parsedState = JSON.parse(state);
+        if (parsedState.description || parsedState.errors) {
+          console.log('error add to cart', parsedState)
+          // Error handling
+          return;
+        }
+
+        this.renderSections(parsedState);
+
+        publish(PUB_SUB_EVENTS.cartUpdate, { source: 'quick-add', cartData: parsedState });
+
+      }).catch((error) => {
+        console.error(error);
+      })
+  }
+
+  getSectionsToRender() {
+    return [
+      {
+        id: `quick-add-bulk-${this.dataset.id}`,
+        section: document.getElementById('product-grid').dataset.id,
+        selector: `#quick-add-bulk-${this.dataset.id}`
+      },
```

Comment:
Working on the progress bar I faced a few difficulties that makes my code more complex because of we re render these elements. That made me questioned the sense of it. 
These elements contain input and buttons. Do we really need to re-render them? At a quick glance all that should be updated there is qty but it is updated anyway without re-rendering.

Do I missing something?

Reply:
Thanks for sharing Eugene! Yes we do, one of reasons for the re-render is because we need the updated cart qty in the qty input `data-cart-quantity`. Otherwise you are updating the value, but not getting the updated element with the updated items in cart

What difficulties are you facing? Could you share a bit more?